test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationParts/AssemblyPartTest.cs

```diff
@@ -22,6 +22,20 @@ public void AssemblyPart_Name_ReturnsAssemblyName()
         }
 
         [Fact]
+        public void AssemblyPart_Types_ReturnsDefinedTypes()
+        {
+            // Arrange
+            var assembly = typeof(AssemblyPartTest).GetTypeInfo().Assembly;
+            var part = new AssemblyPart(assembly);
+
+            // Act
+            var types = part.Types;
+
+            // Assert
+            Assert.Equal(assembly.DefinedTypes, types);
```

Comment:
I believe that's what Equal in xunit does. Maybe I can just new up a collection with DefinedTypes as source to be sure? The purpose of this test is to make sure that something breaks if someone decides to change it later for GetExportedTypes or something like that, to ensure that such change doesn't go unnoticed